src/USER-MISC/pair_kolmogorov_crespi_full.cpp

```diff
@@ -727,7 +742,6 @@ void PairKolmogorovCrespiFull::KC_neigh()
 
     KC_firstneigh[i] = neighptr;
     KC_numneigh[i] = n;
-    if (n == 0) error->all(FLERR,"Could not build neighbor list to calculate normals, please check your configuration");
     if (n > 3) error->all(FLERR,"There are too many neighbors for some atoms, please check your configuration");
```

Comment:
same as above.

Reply:
@akohlmey Thanks for the information. I will do some tests on it.

Reply:
@akohlmey I changed error->all to error->one and did a test for multi-layer graphene, it seems the stuck problem has been resolved. But I still prefer to delete this check for the following reasons:
1, when n=0, the codes set all the normals along the z-direction, which allows comparing with the KC-type potentials that assume all the normals are along the z-direction;
2, the simulation results are not affected by this check.
3, From last year, I was extending these codes to include transition metal dichalcogenides (TMD). I found everything is fine without this check, but if I add this check, the codes stop at this line when I use more than 2 cores to run the simulation, although it runs smoothly on a single CPU. Since the loop goes over owned+ghost atoms when building the neighbour list for calculating the normals, I guess the reason might be that some ghost atoms cannot find neighbours when using multiple cores. However, this is just a guess, I don't really know the reason behind, although delete this check can solve the problem, it will still be great if I can get some information here.

Reply:
> @akohlmey I changed error->all to error->one and did a test for multi-layer graphene, it seems the stuck problem has been resolved. But I still prefer to delete this check for the following reasons:

that is fine by me. i just saw the calls to `Error::all()` and it looked like those were cases, where it is not certain, that all MPI ranks reach this condition. That is just a general issue. It is not always easy to spot. we have had cases, where this was in the code for many years, until somebody by chance exposed it. Or the condition was given when the code was written, but then something else was changed, where this case was no longer given.

> 1, when n=0, the codes set all the normals along the z-direction, which allows comparing with the KC-type potentials that assume all the normals are along the z-direction;
> 2, the simulation results are not affected by this check.
> 3, From last year, I was extending these codes to include transition metal dichalcogenides (TMD). I found everything is fine without this check, but if I add this check, the codes stop at this line when I use more than 2 cores to run the simulation, although it runs smoothly on a single CPU. Since the loop goes over owned+ghost atoms when building the neighbour list for calculating the normals, I guess the reason might be that some ghost atoms cannot find neighbours when using multiple cores.

yes, even with requesting neighbors of ghost atoms, there is a limit to how many neighbors are included, when you reach the proximity of the communication cutoff. those missing atoms are simply not there. neightbors of ghosts are meant to be used for tracking down neighbors of neighbors, e.g. when looking at implicit dihedral interactions. you can increase how far this goes by increasing the communication cutoff. this defaults to maximum pairwise cutoff plus neighbor list skin.

Reply:
Thanks for the helpful information!